cmake/MacroGuardTest.cmake

```
@@ -0,0 +1,107 @@
+# PROJECT_NAME: bsoncxx or mongocxx.
+# GUARDED_MACROS: list of macros that should be guarded by prelude/postlude headers.
+# INCLUDE_FILTERS: list of regex filters for headers to be added to the tests (must be relative to PROJECT_SOURCE_DIR).
+# EXCLUDE_FILTERS: list of regex filters for headers to be excluded (applied after INCLUDE_FILTERS).
+# PROJECT_TEST_PROPERTIES: the (bsoncxx|mongocxx)_test_properties target to be linked with.
```

Use a bracket-comment for easier reading+writing:

```suggestion
#[==[
Define a build-time test that asserts that the given macros are maintained across
the inclusion of the header files that match the given pattern

    add_macro_guard_test(
        PROJECT_NAME <name>
        INCLUDE_FILTERS [pattern...]
        [EXCLUDE_FILTERS [pattern...]]
        GUARDED_MACROS [macro...]
        PROJECT_TEST_PROPERTIES <target>
    )

PROJECT_NAME
    bsoncxx or mongocxx.

GUARDED_MACROS
    list of macros that should be guarded by prelude/postlude headers.

INCLUDE_FILTERS
    list of regex filters for headers to be added to the tests (must be relative to PROJECT_SOURCE_DIR).

EXCLUDE_FILTERS
    list of regex filters for headers to be excluded (applied after INCLUDE_FILTERS).

PROJECT_TEST_PROPERTIES
    the (bsoncxx|mongocxx)_test_properties target to be linked with.
]==]
```